plugin/service/storage/plugin.go

```diff
@@ -159,6 +161,29 @@ func (p *StoragePlugin) call(
 	}
 }
 
+// NormalizeStorageBucketData is called by boundary before calling OnCreate or OnUpdate
+//
+// aws storage supports a total of 8 attributes, 6 as part of credential.CredentialAttributes
+// and 2 more from storage.StorageAttributes. of these, only endpoint_url is normalized.
+func (sp *StoragePlugin) NormalizeStorageBucketData(ctx context.Context, req *pb.NormalizeStorageBucketDataRequest) (*pb.NormalizeStorageBucketDataResponse, error) {
+	sa, err := getStorageAttributes(req.GetAttributes())
+	if err != nil {
+		return nil, err
+	}
+
+	fields := req.GetAttributes().GetFields()
+	if sa.EndpointUrl, err = parseutil.NormalizeAddr(sa.EndpointUrl); err != nil {
```

Comment:
Wouldn't this error if a endpoint url field was not provided? This field is optional in this plugin